### ISSUE TITLE
Ignore specific snapshots from deletion

### ISSUE THREAD

**Issue by dimsanius** [Meru3m/virtualbox-snapshotter#12]
Hello,

I have come across a need to retain specific snapshots even if they are old enough to be deleted.

I want to implement such a feature when I will have time, but I may need some ideas before implementing it:
- Using snapshot names may be sufficient for identification. Are there any better ways to identify snapshots?
- I was thinking of implementing a separate file (like `snapshots.ignore`) to be able to store a list of snapshot identifiers to be retained no matter how old they are. Are there any other options to store a specific snapshot retention list?
  - It would be quite cool to have an extra argument (i.e. `-i`/`--ignore`) that would accept either an aforementioned `snapshots.ignore` or a snapshot identifier straight from terminal. I.e. `python virtualbox_snapshotter.py -i "My awesome snapshot to be retained"`. or `python virtualbox_snapshotter.py -i snapshots.ignore`, where `snapshots.ignore` file contains all the snapshots to be retained.
  
  Let me know what do you think and if you have any suggestions!

**Comment by mattiabaldari** [comment]
@dimsanius I'll reply point by point:
- wouldn't be better to use dates as key plus a description (so there could be multiple snapshot for the same day)?
- I like the `snapshots.ignore` idea and the way you want to use the arguments.

**Comment by dimsanius** [comment]
- I have realised snapshots name and descriptions my not be unique. The only thing that is unique - UUID. But it is not accessible from VirtualBox GUI. It can be accessed from `VBoxManage.exe` CLI. I guess I will have to add a bit extra lines into readme on how to use it to get the UUID. In a nutshell: `VBoxManage.exe snapshot "MACHINE_NAME" list` will give you a full list of snapshots and their UUIDs.
- Thanks! I think of the best way to store UUIDs in a file, along with comments. Comments will allow a better transparency over time. For example file containing this line doesnt make sense:
```
6ee353bb-74c6-459b-a621-ea4c2d50ee2a
6ee353bb-74c6-4f9f-a621-aa4a2d50ee2a
6ee353bb-74c6-4f9f-a621-ea4c2d50de2f
```

But this, makes much more sense:
```
6ee353bb-74c6-459b-a621-ea4c2d50ee2a       # Snapshot name: "My awesome snapshot 1"
6ee353bb-74c6-4f9f-a621-aa4a2d50ee2a       # Snapshot name: "My awesome snapshot 2"
6ee353bb-74c6-4f9f-a621-ea4c2d50de2f       # Snapshot name: "My awesome snapshot 3"
```

So, I want to find the best way of having such a file, with an ability to have comments

**Comment by mattiabaldari** [comment]
basically you are re-creating git commits

**Comment by dimsanius** [comment]
Sorry, was a bit busy.

Yes, similar to git commits. However, instead of commit IDs, it will be snapshot IDs.

**Comment by mattiabaldari** [comment]
@dimsanius don't worry, Software Developer life it's just that ;)
I saw your last PR solved this issue, may I close it?
Probably it is time to tag the new version as well.

**Comment by dimsanius** [comment]
@Meru3m Yes, the issue can now be closed.
And new version feels like a good idea. Maybe `v1.2.0`? Its more of a new feature, then a bugfix.
Changelog is available at #13

**Comment by mattiabaldari** [comment]
Implemented in version v1.2.0
